api/v1alpha1/hostedcluster_types.go

```diff
@@ -324,6 +324,34 @@ type AWSPlatformSpec struct {
 	// The secret should have exactly one key, `credentials`, whose value is
 	// an AWS credentials file.
 	NodePoolManagementCreds corev1.LocalObjectReference `json:"nodePoolManagementCreds"`
+
+	// resourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+	// See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+	// AWS supports a maximum of 50 tags per resource. OpenShift reserves 25 tags for its use, leaving 25 tags
+	// available for the user.
+	// +kubebuilder:validation:MaxItems=25
+	// +optional
+	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
```

Comment:
The format is copied from the openshift api types and different from the capi types, where this is a typed `map[string]string`: https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L361-L381

Not sure which is better and no strong opinion either way.

Reply:
From a UX pov the one you propose seems fine and clear to me and matches ocp. From a technical pov the one you propose I think is probably easier and more granular to validate via kubebuilder tags. So +1 from me.